### DISCUSSION TITLE
How to migrate from v1 credentials ChainProvider

### DISCUSSION THREAD

**Discussion by BananZG** [aws/aws-sdk-go-v2#2130] [Q&A]
Currently in our code we have a list of credentials.Provider defined with V1 SDK, something like:
```
sess, err := session.NewSession(&aws.Config{
    Region: aws.String(region),
})

endpoint := "" // example : some url
profile := "" // example : lambda_role
sess.Config.Credentials = credentials.NewChainCredentials([]credentials.Provider{
    &credentials.EnvProvider{},
    &credentials.SharedCredentialsProvider{
        Profile: profile,
    },
    &ec2rolecreds.EC2RoleProvider{
        Client: ec2metadata.New(sess),
    },
    endpointcreds.NewProviderClient(*sess.Config, sess.Handlers, endpoint)
    // .... and so on
}
```

in this way, we are able to deploy same code to different environments that read credentials in different way.
Locally different devs have different ways to run their code too, some with aws-vault's ecs, some with ec2, some just run the command directly with custom profile.

Now with V2, if I defined all these in one shot, the credentials doesnt 'chain'.
Example when I try to run 
```
cfg, err := config.LoadDefaultConfig(ctx,
    config.WithRegion(region),
    config.WithSharedConfigProfile(profile),
    config.WithCredentialsProvider(aws.NewCredentialsCache(ec2rolecreds.New())),
)
```
it's trying to connect to my ec2 but what I really want to is for it to try connecting with profile first, then with ec2.

So, is there any way to achieve this like the old ChainProvider without having to add on conditions in code?
Thanks!!!

**Comment by avishay28** [comment]
What did you do eventually?
I'm having the same issue

**Comment by chris-wilson-r7** [comment]
Was anybody able to achieve the same thing in v2?
